ilastik/applets/dataSelection/dataSelectionGui.py

```diff
@@ -515,21 +505,60 @@ def _determineLaneRange(self, infos: List[DatasetInfo], startingLaneNum=None):
 
         return (startingLaneNum, endingLane)
 
-    def _createDatasetInfos(self, roleIndex: int, filePaths: List[Path], rois=None):
-        """
-        Create a list of DatasetInfos for the given filePaths and rois
-        rois may be None, in which case it is ignored.
-        """
-        if rois is None:
-            rois = [None] * len(filePaths)
-        assert len(rois) == len(filePaths)
+    def _join_dataset_path(self, file_path: Path, internal_path: str) -> str:
+        """Return a proper path to the dataset in a file."""
+        p = file_path.absolute()
+        with contextlib.suppress(ValueError):
+            p = p.relative_to(self.topLevelOperator.WorkingDirectory.value)
```

Comment:
probably unrelated comment: I will always stumble over this `WorkingDirectory` thing being the project file dir...